helix-core/src/test/java/org/apache/helix/manager/zk/TestZkBaseDataAccessor.java

```diff
@@ -188,6 +199,33 @@ public void testSyncCreate() {
     System.out.println("END " + testName + " at " + new Date(System.currentTimeMillis()));
   }
 
+  @Test
+  public void testSyncCreateWithCustomSerializer() {
```

Comment:
How did you verify the dataAccessor actually uses the serializer given by the caller, not the default one?

Reply:
The updated test takes a list serializer/deserilaizer as the input parameter, it's different from default one. Do you think we need to test the constructor if it really overrides the zkClient with custom serializer?

Reply:
Working with @i3wangyi to add more test cases to test this.